platform/src/main/java/org/hillview/storage/CassandraDatabase.java

```diff
@@ -0,0 +1,264 @@
+    private static final String ssTableNameMarker = "big-Data.db";
+    /** The default path of Cassandra's sstableutil tool */
+    private static final String ssTableUtilPathUnix = "/bin/sstableutil";
+    private static final String ssTableUtilPathWindows = "\bin\sstableutil";
+
+    public CassandraConnectionInfo info;
+    // Session: Connection to cassandra cluster for executing client-side queries
+    @Nullable
+    private Cluster cluster;
+    @Nullable
+    private Session session;
+    @Nullable
+    private Metadata metadata;
+    // Probe: Connection to cassandra node that enables Hillview to execute server-side query
+    @Nullable
+    private INodeProbeFactory nodeProbeFactory;
+    @Nullable
+    private NodeProbe probe;
+    private List<CassTable> cassTables = new ArrayList<CassTable>();
+    private List<TokenRange> tokenRanges = new ArrayList<TokenRange>();
+
+    /** This class enables Hillview to execute server-side and client-side queries, such as
+     * getting sstable path, list of stored tables, list of token-range, and also the row count
+     * of a specific table.
+     * The flow of execution to locate a set of table partitions is as follows:
+     * - CassandraDatabase establishes NodeProbe connection to get full token-range and its endpoints
+     * - CassandraDatabase find the sstable path by running sstableutil
+     * - Finally, CassandraSSTableReader uses both output to read the local token-range
+    */
+    public CassandraDatabase(CassandraConnectionInfo info) {
+        this.info = info;
+        try {
+            this.nodeProbeFactory = new NodeProbeFactory();
+            this.connectLocalProbe();
+            this.connectCassCluster();
+            this.setStoredTableInfo();
+            this.setTokenRanges();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /** Connection for executing client query. */
+    private void connectCassCluster() {
+        Builder b = Cluster.builder().addContactPoint(info.host);
+        b.withPort(info.port);
+
+        if (!Utilities.isNullOrEmpty(info.user)) {
+            b.withCredentials(info.user, info.password);
+        }
+
+        this.cluster = b.build();
+        this.session = this.cluster.connect();
+        this.metadata = this.cluster.getMetadata();
+    }
+
+    public void closeClusterConnection() {
+        if (this.session != null)
+            this.session.close();
+        if (this.cluster != null)
+            this.cluster.close();
+    }
+
+    /** Connection for running server-side query. */
+    private void connectLocalProbe() {
+        try {
+            if (Utilities.isNullOrEmpty(info.user)) {
+                this.probe = nodeProbeFactory.create(info.host, info.jmxPort);
+            } else
+                this.probe = nodeProbeFactory.create(info.host, info.jmxPort,
+                    info.user, info.password);
+        } catch (IOException | SecurityException e) {
```

Comment:
instead of catching the exceptions here add them to the method signature and let the caller catch them.